config/routes.rb

```diff
@@ -401,6 +401,15 @@
           put 'test_connection'
         end
         resources :images, :except => [:show]
+        resources :key_pairs, :only => [:index] do
+          member do
+            get 'download'
+          end
+          collection do
+            put 'recreate'
+            delete 'remove'
```

Comment:
this assumes you can only remove the compute resource key? I think we should support removing other keys as well. in that case you would still need to pass the key id.